### DISCUSSION TITLE
How to dispose of individual elements from hierarchy? (eg. Label, AbsoluteLayout, Border, Image, Editor)

### DISCUSSION THREAD

**Discussion by jonmdev** [dotnet/maui#21251] [Q&A]
Let's say you create a hierarchy like:

```
AbsoluteLayout abs= new();
Border border = new();
Label label = new();
abs.Add(border);
abs.Add(label);

existingRootLayoutElement.Add(abs);
```

Then you want to discard of those elements. You can go:

```
abs.Remove(border);
abs.Remove(label);
existingRootLayoutElement.Remove(abs);
```

To fully dismantle the hierarchy. But then how do you make those elements get destroyed/disposed of in the native system? Does Maui have some way of saying: "This element is no longer needed and can be garbage collected"?

Or do we need a custom native way of handling this in each OS?

Otherwise, every element you create will keep taking up memory after it's no longer needed.

For reference, I am not using any standard NavigationStack or any of that stuff. I am just building my own hierarchies as needed from basic elements. The only things I use are Label, AbsoluteLayout, Border, Image, & Editor. 

Any thoughts? Thanks.

**Comment by pme442** [comment]
I also have this question.

**Comment by AdamEssenmacher** [comment]
https://github.com/dotnet/maui/issues/21809#issuecomment-2053759416
